### PR TITLE
build(sequence): use fluid-tsc

### PR DESCRIPTION
**Pull request by DLehenbauer**
Semi-automatic PR to switch to fluid-tsc

### CODE REVIEW

**Review by sonalideshpandemsft — APPROVED** [review]

File: packages/dds/sequence/package.json [hunk 1]
```diff
@@ -97,7 +99,7 @@
 		"lint": "npm run prettier && npm run check:release-tags && npm run eslint",
 		"lint:fix": "npm run prettier:fix && npm run eslint:fix",
 		"postpack": "tar -cf ./sequence.test-files.tar ./src/test ./dist/test",
-		"perf": "cross-env FLUID_TEST_VERBOSE=1 mocha \"dist/**/*.spec.*js\" --node-option unhandled-rejections=strict,expose-gc --exit -r node_modules/@fluid-internal/mocha-test-setup --perfMode --fgrep @Benchmark --reporter @fluid-tools/benchmark/dist/MochaReporter.js --timeout 30000",
+		"perf": "cross-env FLUID_TEST_VERBOSE=1 mocha \"dist/**/*.spec.*js\" --node-option unhandled-rejections=strict,expose-gc --exit --perfMode --fgrep @Benchmark --reporter @fluid-tools/benchmark/dist/MochaReporter.js --timeout 30000",
```
Comment (jason-ha) [inline packages/dds/sequence/package.json:102]: test ESM?

Comment (jason-ha) [reply packages/dds/sequence/package.json:102]: +5 more instances

File: packages/dds/sequence/src/test/tsconfig.json [hunk 2]
```diff
@@ -1,18 +1,16 @@
 {
-	"extends": ["../../../../../common/build/build-common/tsconfig.test.json"],
+	"extends": "../../../../../common/build/build-common/tsconfig.test.node16.json",
+	"compilerOptions": {
+		"rootDir": "./",
+		"outDir": "../../lib/test",
+		"types": ["mocha", "node"],
+		"noImplicitAny": false,
+		"noUnusedLocals": false, // Need it so memory tests can declare local variables just for the sake of keeping things in memory,
```
Comment (jason-ha) [inline packages/dds/sequence/src/test/tsconfig.json:8]: Pre-existing - better to have these handled explicitly where they are. Not suggesting change in this PR.

**Review by jason-ha — CHANGES_REQUESTED** [review]

File: azure/packages/test/end-to-end-tests/package.json [hunk 3]
```diff
@@ -38,12 +39,12 @@
 		"cache-dir": "nyc/.cache",
 		"exclude": [
 			"src/test/**/*.*ts",
```
Comment (jason-ha) [inline azure/packages/test/end-to-end-tests/package.json:41]: pre-existing, but this coverage setup config seems odd if not incorrect. The only files in this package are test files. No need to change, but I'd be curious to understand why we bother with coverage of nothing... or if that is not the effect.?

File: packages/dds/sequence/src/test/tsconfig.json (hunk 2 above)
Comment (jason-ha) [reply packages/dds/sequence/src/test/tsconfig.json:8]: AB#7423

File: azure/packages/test/end-to-end-tests/package.json (hunk 3 above)
Comment (jason-ha) [reply azure/packages/test/end-to-end-tests/package.json:41]: [AB#7424](https://dev.azure.com/fluidframework/internal/_workitems/edit/7424)

**Review by jason-ha — APPROVED** [review]

**Review by jason-ha — APPROVED** [review]

**Review by anthony-murphy — APPROVED** [review]

**Review by Abe27342 — APPROVED** [review]

**Pull request merged**